### PR TITLE
improve publish new releases process

### PR DESCRIPTION
**Pull request by ezypeeze**
@rchl let me know your thoughts about this, would be nice to have something that didn't require to push to master directly I guess.

PS: About the deploy_docs.sh change, it's just because I use alot of config ssh v-hosts. It will pick the same one as the registed remote.

### CODE REVIEW

**Comment by rchl** [comment]
I don't fully understand this change.

I guess now the release script will create a new commit and tag locally but nothing will be released until those are pushed still.

Is this meant to be a starting point for a fix rather than a complete solution? If so then a solution could be to add some GitHub action that creates and updates a release PR on every new change on master. Then a merge of that PR would trigger a release. That would also be good for visibility on what is going into a release.

I haven't implemented that kind of flow before though so that would need some investigation

**Comment by ezypeeze** [comment]
> I don't fully understand this change.
> 
> I guess now the release script will create a new commit and tag locally but nothing will be released until those are pushed still.
> 
> Is this meant to be a starting point for a fix rather than a complete solution? If so then a solution could be to add some GitHub action that creates and updates a release PR on every new change on master. Then a merge of that PR would trigger a release. That would also be good for visibility on what is going into a release.
> 
> I haven't implemented that kind of flow before though so that would need some investigation

"Is this meant to be a starting point for a fix rather than a complete solution?"
Yep, it's to investigate a better way to make releases, would love a github action button to do it (and also run this deploy_docs script, which basically pushs docs to gh-pages branch)

**Comment by rchl** [comment]
That's could useful...

We would have an input where we would have to type the type of the release (major, minor, patch) then? I guess that would be fine.

But since the project is following the Conventional Commits convention, the decision whether the release should be major, minor or patch could be made automatically. For example I'm using https://github.com/release-it/release-it in some projects that does that.

File: deploy_docs.sh [hunk 1]
```diff
@@ -5,6 +5,12 @@
 
 GIT_NAME=$(git config --get user.name)
 GIT_EMAIL=$(git config --get user.email)
+GIT_REMOTE_URL=$(git config --get remote.origin.url)
```
Comment (rchl) [inline deploy_docs.sh:8]: Not sure this will work on CI. I think CI might be checking out detached HEAD.

**Review by rchl — APPROVED** [review]
Nice and simple.

**Comment by ezypeeze** [comment]
> That could be useful...
> 
> We would have an input where we would have to type the type of the release (major, minor, patch) then? I guess that would be fine.
> 
> But since the project is following the Conventional Commits convention, the decision whether the release should be major, minor or patch could be made automatically. For example I'm using https://github.com/release-it/release-it in some projects that does that.

I would still prefer to manually decide the type of release, because if we by mistake add a "!" or something, it will assume breaking changes and will release a major unecessary. But correct me if I'm wrong.

I think I did it, the idea was having 3 workflows, one for each type, that will on run manually and for master branch.
Will do the new version, push the changes to master, publish and finally deploy the docs.

**PS:**  I'm not sure if github action will be able to push to master out-of-the-box.

File: deploy_docs.sh (hunk 1 above)
Comment (rchl) [reply deploy_docs.sh:8]: Actually I can see that I'm just doing `git push` in [another workflow](https://github.com/resoai/TileBoard-addon/blob/b6a67d80c82edb634503110b31de91ed6e1bbb79/.github/workflows/on_tileboard_release.yaml#L37-L37) so it should be working.

The only difference I also have `fetch-depth: 0  # with tags` set for the checkout step.

**Comment by ezypeeze** [comment]
@rchl does github actions have fail-fast as default? like if pushing to master step fails, will not procceed to next step?

**Comment by rchl** [comment]
Yes, it should work like that, as long as the command returns non-0 exit code.

**Pull request merged**